src/components/content/catalog/services/tree/ServiceTree.tsx

```diff
@@ -18,15 +20,60 @@ export function ServiceTree({
     setSelectKey: (selectedKey: React.Key) => void;
     isViewDisabled: boolean;
 }): React.JSX.Element {
-    function isParentTreeSelected(selectKey: React.Key): boolean {
+    const [searchValue, setSearchValue] = useState('');
+
+    const filteredTreeData = useMemo(() => {
+        const filterNodes = (nodes: DataNode[]): DataNode[] => {
+            return nodes
+                .map((node) => {
+                    const matches =
+                        (node.title?.toString().toLowerCase() ?? '').includes(searchValue.toString().toLowerCase()) ||
+                        node.key.toString().toLowerCase().includes(searchValue.toString().toLowerCase());
+
+                    if (!node.children) {
+                        return matches ? node : null;
+                    }
+                    const filteredChildren = filterNodes(node.children);
+                    return filteredChildren.length > 0 || matches ? { ...node, children: filteredChildren } : null;
+                })
+                .filter(Boolean) as DataNode[];
+        };
+
+        return filterNodes(treeData);
+    }, [searchValue, treeData]);
+
+    useMemo(() => {
+        if (filteredTreeData.length > 0) {
+            const findFirstLeaf = (nodes: DataNode[]): React.Key | null => {
+                for (const node of nodes) {
+                    const { key, children } = node;
+                    if (children && children.length > 0) {
+                        const found = findFirstLeaf(children);
+                        if (found) {
+                            return found;
+                        }
+                    } else {
+                        return key;
+                    }
+                }
+                return null;
+            };
+            const firstLeafKey = findFirstLeaf(filteredTreeData);
+            if (firstLeafKey) {
+                setSelectKey(firstLeafKey);
```

Comment:
there is a warning here

![image](https://github.com/user-attachments/assets/620dfb01-1049-424e-ab8b-a794add7dc5c)